### ISSUE TITLE
Ruby unable to build native extensions

### ISSUE THREAD

**Issue by That-Human-Being** [msys2/MINGW-packages#12788]
Whenever I try to install Gems such as Rails or Devel I get this error:
```
Building native extensions. This could take a while...
ERROR:  Error installing devel:
        ERROR: Failed to build gem native extension.

    current directory: /home/user/.gem/ruby/2.7.0/gems/nokogiri-1.13.8/ext/nokogiri
/usr/bin/ruby.exe -I /usr/lib/ruby/site_ruby/2.7.0 extconf.rb
checking for whether -std=c99 is accepted as CFLAGS... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/usr/bin/$(RUBY_BASE_NAME)
        --help
        --clean
        --prevent-strip
/usr/lib/ruby/2.7.0/mkmf.rb:472:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
        from /usr/lib/ruby/2.7.0/mkmf.rb:598:in `block in try_compile'
        from /usr/lib/ruby/2.7.0/mkmf.rb:545:in `with_werror'
        from /usr/lib/ruby/2.7.0/mkmf.rb:598:in `try_compile'
        from /usr/lib/ruby/2.7.0/mkmf.rb:662:in `try_cflags'
        from /usr/lib/ruby/2.7.0/mkmf.rb:668:in `block (2 levels) in append_cflags'
        from /usr/lib/ruby/2.7.0/mkmf.rb:972:in `block in checking_for'
        from /usr/lib/ruby/2.7.0/mkmf.rb:362:in `block (2 levels) in postpone'
        from /usr/lib/ruby/2.7.0/mkmf.rb:332:in `open'
        from /usr/lib/ruby/2.7.0/mkmf.rb:362:in `block in postpone'
        from /usr/lib/ruby/2.7.0/mkmf.rb:332:in `open'
        from /usr/lib/ruby/2.7.0/mkmf.rb:358:in `postpone'
        from /usr/lib/ruby/2.7.0/mkmf.rb:971:in `checking_for'
        from /usr/lib/ruby/2.7.0/mkmf.rb:667:in `block in append_cflags'
        from /usr/lib/ruby/2.7.0/mkmf.rb:666:in `each'
        from /usr/lib/ruby/2.7.0/mkmf.rb:666:in `append_cflags'
        from extconf.rb:617:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/user/.gem/ruby/2.7.0/extensions/x86_64-unknown/2.7.0/nokogiri-1.13.8/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /home/user/.gem/ruby/2.7.0/gems/nokogiri-1.13.8 for inspection.
Results logged to /home/user/.gem/ruby/2.7.0/extensions/x86_64-unknown/2.7.0/nokogiri-1.13.8/gem_make.out
```

I tested to see what would happen on Fedora Linux and all I needed to do there was install `ruby-devel`, Arch Linux bundles the necessary headers as `ruby-stdlib` and is included as a part of the Ruby package. Could this be fixed?

**Comment by Biswa96** [comment]
It seems that you are using msys2 ruby package. Would you like to try the mingw ruby? See https://packages.msys2.org/base/mingw-w64-ruby

**Comment by That-Human-Being** [comment]
I just entered 
```$ gem install devel```
mingw ruby didn't work either

**Comment by That-Human-Being** [comment]
So I found the issue is probably with a dependancy of Rails [nio4r](https://github.com/socketry/nio4r)

**Comment by thomas-henley** [comment]
@That-Human-Being how did you fix it? I am having the same issue installing puma.

**Comment by Biswa96** [comment]
Would you like to provide proper steps to reproduce the problem?

**Comment by thomas-henley** [comment]
My problem was on Windows but looked almost identical to the original post. I think I worked out the cause in my case, and was able to resolve it by just reinstalling Ruby. I had set up my environment by installing Ruby with Chocolatey (`choco install ruby`), but that left out important dev kit tools needed for `nio4r` and `puma`. I tried a bunch of different things but kept running into issues -- problems because I didn't have `make`, `gcc`, etc. 

The fix was to remove my chocolatey Ruby and install Ruby with a proper devkit from https://rubyinstaller.org/downloads/. That one worked without issue right out of the box.
